rest_framework-stubs/validators.pyi

```diff
@@ -1,10 +1,17 @@
-from typing import Any, Container, Iterable, MutableMapping, NoReturn, Optional, TypeVar
+from typing import Any, Container, Iterable, MutableMapping, Optional, TypeVar, Protocol, Union, Callable
 
 from django.db.models import Model, QuerySet
 from rest_framework.fields import Field
 from rest_framework.serializers import BaseSerializer
 
 _T = TypeVar("_T", bound=Model)
+_V = TypeVar("_V", contravariant=True)
+
+class ContextValidator(Protocol[_V]):
+    requires_context: bool
+    def __call__(self, __value: _V, __context: Field) -> None: ...
+
+Validator = Union[Callable[[_V], None], ContextValidator[_V]]
```

Comment:
`ContextValidator` and `Validator` are symbols that don't exist in DRF code proper. Should I start them with underscore or keep as is?

There already are a few examples of stubs-only classes without underscore, such as `AsView`, `ViewSetAction`, `UsesQuerySet`.

Reply:
Are these items supposed to be used by the end user? Or only by our own code? 

If they are public, then we would have to make them public and also provide some kind of a runtime fallback.

Reply:
I don't expect end-users to need to reference them, unless they implement their own code that accepts validators.

For AsView, ViewSetAction I've used this trick:
```python
if TYPE_CHECKING:
    # Only defined in djangorestframework-stubs
    from rest_framework.decorators import ViewSetAction
    from rest_framework.views import AsView

class MyViewSet(GenericViewSet):
    @classonlymethod
    def as_view(cls, actions: "Optional[Dict[str, Union[str, ViewSetAction]]]" = None, **kwargs: Any) -> "AsView":
```

By importing when `TYPE_CHECKING` is set and quoting them, it doesn't break runtime.